### PR TITLE
Use thread::sleep instead of deprecated sleep_ms

### LINKED ISSUE #8694: Remove warnings on using deprecated std::thread::sleep_ms
When building servo now, there is a lot of warning messages in the following form:
```
/home/keith/Workspace/servo/ports/glutin/window.rs:301:13: 301:34 warning: use of deprecated item: replaced by `std::thread::sleep`, #[warn(deprecated)] on by default
/home/keith/Workspace/servo/ports/glutin/window.rs:301         use std::thread::sleep_ms;
                                                                   ^~~~~~~~~~~~~~~~~~~~~
```
We'll have to update our definitions to use the new `std::thread::sleep`, which takes a `Duration` struct instead of a `u32`.

Code: `components/profile/mem.rs`, `components/profile/time.rs`, `components/webdriver_server/lib.rs`, `components/devtools/actors/timeline.rs`, `components/compositing/scrolling.rs`, `ports/glutin/window.rs`
Documentation: https://doc.rust-lang.org/nightly/std/thread/fn.sleep.html, https://doc.rust-lang.org/nightly/std/time/struct.Duration.html

### PR DESCRIPTION
**Pull request by jsanders**
Similarly, change one instance of `thread::park_timeout_ms`.

Fixes #8694

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8705)

### CODE REVIEW

**Comment by metajack** [comment]
A small refactoring is suggesting. Otherwise r=me.

---

Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4BHnqZPSRdl8D5RRz3:570197562) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
Can we get rid of this duplicated code by refactoring this logic into a helper function?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705#-:-K4BHxmqTuVPnUHHCfv3:2070515030)*

**Comment by jsanders** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4DwVA4ABbhMVTmewWy:1263416748) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I'd love to do that, but I wasn't sure where something like that would go. I was considering `util::task` or making a new `util` module in `profile`.  Any suggestions?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*

**Comment by Wafflespeanut** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4EhlqkhkyDk7_XQ5Wi:-106037974) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
`util::task` is currently being used for spawning threads, IIRC. I guess it'd be better if we just create a new module in `util` (say, `time.rs`), and put our function there.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*

**Comment by jsanders** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4EjiXbA1H4ZGP2iFOC:1396299650) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I also looked into parsing the profiler options directly into `Some<Duration>` instead of `Some<f64>`, but that won't work unless I add a `serde::Serializer` implementation, which I'm not sure is the right approach.

Adding `util::time` seems reasonable.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*

**Comment by jsanders** [comment]
Review status: 6 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[components/profile/time.rs, line 126 \[r1\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4BHnqZPSRdl8D5RRz2:-K4ErZnH541kzaRs0tLa:257465616) ([raw file](https://github.com/servo/servo/blob/2cb0887a627c4f9a667bf74d4bbbc6509c525b8a/components/profile/time.rs#L126)):*
I went ahead and pulled this logic out as `util::time::duration_from_seconds`.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705)*

**Comment by Wafflespeanut** [comment]
@jsanders sgtm, squash the commits, btw?

File: components/compositing/scrolling.rs [hunk 1]
```diff
@@ -55,9 +57,13 @@ impl ScrollingTimerProxy {
...
-            let target = timestamp as i64 + TIMEOUT;
-            let delta_ns = target - (time::precise_time_ns() as i64);
-            sleep_ms((delta_ns / 1000000) as u32);
+            const NANOS_PER_SEC: u64 = 1_000_000_000;
+            let target = timestamp + TIMEOUT;
+            let delta_ns = target - (time::precise_time_ns());
+            let delta_secs = delta_ns / NANOS_PER_SEC;
+            let delta_nanos = delta_ns % NANOS_PER_SEC;
+            assert!(delta_nanos <= u32::MAX as u64);
+            thread::sleep(Duration::new(delta_secs, delta_nanos as u32));
```
Comment (Wafflespeanut) [inline components/compositing/scrolling.rs]: I think we could also have a function for this in `util/time` (for future usage), which takes nanoseconds as input and returns a duration.

File: components/util/time.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::time::Duration;
+use std::{u32, u64};
+
+pub fn duration_from_seconds(secs: f64) -> Duration {
+    const NANOS_PER_SEC: f64 = 1_000_000_000.0;
```
Comment (Wafflespeanut) [inline components/util/time.rs]: We could just pull the constant out, so that it'll be useful for future methods.

**Comment by Wafflespeanut** [comment]
Once those changes have been made, squash the commits, and let's happily land this :)

File: components/compositing/scrolling.rs (hunk 1 above)
Comment (jsanders) [inline components/compositing/scrolling.rs]: Good idea. Note that this code is also very similar to what `Duration::new` already does to normalize the given nanos to be sub-second, but it only takes a `u32`, which `delta_ns` might not fit into.

File: components/util/time.rs (hunk 2 above)
Comment (jsanders) [inline components/util/time.rs]: I wondered about that. It exists in `Duration` itself (well, as a `u32` rather than an `f64`), but it isn't public.

**Comment by jsanders** [comment]
I'm not familiar with your squashing policy. Should I squash all of the commits, or just the ones that I messed up by leaving the file out?

**Comment by KiChjang** [comment]
@jsanders See https://github.com/servo/servo/wiki/Github-&-Critic-PR-handling-101. Ignore everything that has to do with critic; it's an old review system that we don't use anymore.

**Comment by jsanders** [comment]
@KiChjang: Thanks.

**Comment by jsanders** [comment]
@Wafflespeanut: Moved the nanoseconds-to-duration logic to `util::time`, fixed a couple off-by-one errors in the assertions in `duration_from_seconds`, and squashed.

**Comment by Wafflespeanut** [comment]
Awesome! Thanks!

@Manishearth r+ this?

---

Reviewed 2 of 4 files at r4.
Review status: 6 of 10 files reviewed at latest revision, 2 unresolved discussions.

---

*[components/util/time.rs, line 9 \[r3\]](https://reviewable.io:443/reviews/servo/servo/8705#-K4Ezvv1I1RVB0Us2h-Q-r3-9:-K4LD4gzXXzYybXVPxnj:-1434150410) ([raw file](https://github.com/servo/servo/blob/a8e6aef1b1114895099b526f1c2744451d461c4f/components/util/time.rs#L9)):*
That's okay. This will be useful :)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8705#-:-K4LDFf3zaULSVDuPxFd:796626617)*

**Pull request merged**